### PR TITLE
Color interpolation improvements

### PR DESCRIPTION
**Pull request by graszka22**
## Summary

This PR adds some improvements to interpolating colors:
- takes into account gamma correction when interpolating rgb
- makes interpolating in rgb color space the default
- adds corrected hsv interpolation
- fixes a bug with hsv interpolation (`interpolateColors` with hsv didn't work at all)
- adds docs and examples

## Test plan

I added an example in the docs and in the example app. I also added some unit tests.

### CODE REVIEW

File: __tests__/InterpolateColor.test.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import Animated, {
+  interpolateColor,
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+} from '../src';
+
+describe('colors interpolation', () => {
+  it('interpolates rgb without gamma correction', () => {
+    const colors = ['#105060', '#609020'];
```
Comment (tomekzaw) [inline __tests__/InterpolateColor.test.js:13]: Let's add a few more tests:
1. CSS colors (i.e. from `red` to `blue`)
2. Semi-transparent colors (i.e. alpha != 1)

File: docs/docs/api/animations/withTiming.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ Currently supported formats are:
 - `"#rgba"`
 - `"#rrggbb"`
 - `"#rrggbbaa"`
+- css colors like `"red"` or `"blanchedalmond"` (full list here: https://www.w3.org/TR/css-color-3/#svg-color)
```
Comment (tomekzaw) [inline docs/docs/api/animations/withTiming.md:26]:
```suggestion
- CSS colors like `"red"` or `"blanchedalmond"` (full list here: https://www.w3.org/TR/css-color-3/#svg-color)
```

File: docs/docs/api/miscellaneous/interpolateColors.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+| Options                      | Default | Description                                                                                                                                                                                                                                                 |
+| ---------------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| gamma                        | 2.2     | Gamma parameter used in gamma correction.                                                                                                                                                                                                                   |
+| useCorrectedHSVInterpolation | true    | Sometimes (for example when interpolating from yellow to purple) HSV interpolation goes through many other hues. This option allows to reduce the number of hues in such cases by treating hsv hues like a circular spectrum and choosing the shortest arc. |
```
Comment (tomekzaw) [inline docs/docs/api/miscellaneous/interpolateColors.md:34]:
```suggestion
| useCorrectedHSVInterpolation | true    | Sometimes (for example when interpolating from yellow to purple) HSV interpolation goes through many other hues. This option allows to reduce the number of hues in such cases by treating HSV hues like a circular spectrum and choosing the shortest arc. |
```

File: docs/docs/api/miscellaneous/interpolateColors.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+Value from within the input range that should be mapped to a value from the output range.
+
+#### `input range` [Float[]]
+
+An array of Floats that contains points that indicate the range of the input value. Values in the input range should be increasing.
+
+#### `output range` [(string | number)[]]
+
+An array of colors (strings like `'red'`, `'#ff0000'`, `'rgba(10, 10, 10, 1)'` etc.) that contains points that indicate the range of the output value. It should have at least the same number of points as the input range.
```
Comment (tomekzaw) [inline docs/docs/api/miscellaneous/interpolateColors.md:21]:
```suggestion
An array of colors (strings like `'red'`, `'#ff0000'`, `'rgba(255, 0, 0, 0.5)'` etc.) that contains points that indicate the range of the output value. It should have at least the same number of points as the input range.
```

**Review by tomekzaw — APPROVED** [review]
Well done! 👏 

wow

**Review by tomekzaw — APPROVED** [review]

**Comment by tomekzaw** [comment]
Unfortunately, there seems to be some little bug when interpolating using HSV with correction:

Zrzut ekranu 2023-01-3 o 18 02 43

Each first and each last square should have the same color.

For each test case we have now, can we add two more asserts that confirm that the interpolated color for progress 0.0 and 1.0 equals the start color and the end color, respectively?

**Comment by graszka22** [comment]
That was an issue with the widget in the docs. I added the tests just in case and fixed the widget.

**Comment by piaskowyk** [comment]
The `interpolate` method requires an array with more than one element as an `output` parameter but sometimes receives a one-element array.

Screenshot 2023-01-05 at 13 57 45

The video contains steps on how to reproduce the issue

https://user-images.githubusercontent.com/36106620/210785592-d4197806-99b6-4f56-8376-4e0f50a70d9f.mp4

File: Example/src/ColorInterpolationExample.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+) {
+  'worklet';
+  return interpolateColor(progress, [0, 1], [color1, color2], 'HSV');
+}
+
+export function ColorInterpolationExample() {
+  const [color1, setColor1] = useState('#ff0000');
+  const [color2, setColor2] = useState('#00ffff');
+  const [color1Text, setColor1Text] = useState('#ff0000');
+  const [color2Text, setColor2Text] = useState('#ff0000');
```
Comment (piaskowyk) [inline Example/src/ColorInterpolationExample.tsx:146]: Could we store the same initial value in `color2` and `color2Text`?

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**